proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1790 @@
+
+    * Let `named-channels` be a map of channel names defined by `space`, and
+      their corresponding values in `color`.
+
+    * Set `channels` to the result of `map.merge(channels, named-channels)`.
+
+  * Let `value` be the result of calling `map.get(channels, $channel)`.
+
+  * If `value` is `null`, throw an error.
+
+  * Otherwise, return `value`.
+
+
+## Modified Color Module Functions
+
+### `color.hwb()`
+
+These functions are now deprecated. Authors should use global `hwb()` instead.
+
+> Channel clamping and scaling have been removed from the global function,
+> since we now allow out-of-gamut color-channels to be stored as specified.
+
+* ```
+  hwb($channels)
+  ```
+
+  * Return the result of calling the global function `hwb($channels)`.
+
+* ```
+  hwb($hue, $whiteness, $blackness, $alpha: 1)
+  ```
+
+  * Return the result of calling the global function
+    `hwb($hue $whiteness $blackness / $alpha)`.
+
+### `color.mix()`
+
+```
+mix($color1, $color2,
+  $weight: 50%,
+  $method: null)
+```
+
+* If either `$color1` or `$color2` is not a color with a [known color space],
+  throw an error.
+
+* If `$method` is null:
+
+  * If either `$color1` or `$color2` is not a [legacy color], throw an error.
+
+    > Method is required for non-legacy colors. This matches the `color-mix()`
+    > function defined in [Colors Level 5][color-5], and allows us to add
+    > additional default behavior in the future.
+
+  * Let `color1` and `color2` be the result of [converting] `$color1` and
+    `$color2` respectively into the `rgb` [known color space].
+
+  * Let `weight-scale` be the result of [percent-converting] `$weight` with a
+    `max` of 1, and clamping the value between 0 and 1, inclusive.
+
+  * Let `normal-weight` be `weight-scale * 2 - 1`.
+
+  * Let `alpha1` and `alpha2` be the alpha values of `color1` and `color2`,
+    respectively.
+
+  * Let `alpha-distance` be `alpha1 - alpha2`.
+
+  * Let `weight-by-distance` be `normal-weight * alpha-distance`.
+
+  * If `weight-by-distance == -1`, let `combined-weight1` be `normal-weight`.
+
+  * Otherwise:
+
+    * Let `weight-distance-sum` be `normal-weight + alpha-distance`.
+
+    * Let `combined-weight1` be `weight-distance-sum / (1 + weight-by-distance)`.
+
+  * Let `weight1` be `(combined-weight1 + 1) / 2`.
+
+  * Let `weight2` be `1 - weight1`.
+
+  * Let `red1` and `red2` be the red channels of `color1` and `color2`
+    respectively.
+
+  * Let `red` be the result of rounding `red1 * weight1 + red2 * weight2` to
+    the nearest integer.
+
+  * Let `green1` and `green2` be the green channels of `color1` and `color2`
+    respectively.
+
+  * Let `green` be the result of rounding `green1 * weight1 + green2 * weight2`
+    to the nearest integer.
+
+  * Let `blue1` and `blue2` be the blue channels of `color1` and `color2`
+    respectively.
+
+  * Let `blue` be the result of rounding `blue1 * weight1 + blue2 * weight2` to
+    the nearest integer.
```

Comment:
We don't need to do this rounding anymore, since it'll make the result strictly less accurate.